go/enclave/rollupchain/rollup_chain.go

```diff
@@ -207,8 +207,7 @@ func (rc *RollupChain) updateState(b *types.Block) (*obscurocore.BlockState, []*
 
 	rc.storage.SaveNewHead(bs, head, receipts, logs)
 
-	subscribedLogs := rc.subscriptionManager.FilteredSubscribedLogs(logs, head.Header.Hash())
```

Comment:
This is the main change - the subscribed logs are not calculated on every step of `updateState`, but only afterwards when creating a block submission response.